### PR TITLE
libcdio, bump version

### PR DESCRIPTION
**Pull request by Begasus**

### CODE REVIEW

File: dev-libs/libcdio/patches/libcdio-2.0.0.patchset [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+diff --git a/configure.ac b/configure.ac
+index f8f7f7d..dc352ab 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -260,7 +260,7 @@ AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
+ dnl empty_array_size
+ AC_MSG_CHECKING([how to create empty arrays])
+ 
+-empty_array_size="xxxx"
++empty_array_size="xxx"
```
Comment (Begasus) [inline dev-libs/libcdio/patches/libcdio-2.0.0.patchset]: For some reason this doesn't work when trying to build vcdimager latest version ....
/packages/libcdio-2.0.0-1/.self/develop/headers/cdio/rock.h:226: `xxxx' undeclared here (not in a function)

Comment (alaviss) [reply dev-libs/libcdio/patches/libcdio-2.0.0.patchset]: Clean your work folder and try again? If it still doesn't work then can you post the entire build log?

**Comment by scottmc** [comment]
@Begasus is this tested and works and ready for merge?  If so, then let's have @fbrosson check it merge if ok.

**Comment by Begasus** [comment]
Somehow mixed up the old recipe fix with the bump, will update in a bit if it's ok

**Comment by Begasus** [comment]
updated, if needed the old recipe can be removed, test fails on x86_gcc2 and x86 (secondaryArch) for realpath (on both versions) ...

**Comment by Begasus** [comment]
libcddb could be updated to use this one too ...

**Comment by Begasus** [comment]
Checked on all arch's, builds ok, test fail on all 3 arch's for realpath, this is needed to be able to build the latest vcdimager recipe: https://github.com/haikuports/haikuports/pull/2780

**Comment by scottmc** [comment]
Does the resulting package work ok with vcdimager?  or does the realpath issue cause it problems?  Did the previous version of libcdio also fail this same test?  If so please open a new haikuports issue for it, since Haiku "should" support realpath.

**Comment by Begasus** [comment]
Issue for realpath was already reported: https://github.com/haikuports/haikuports/issues/2777
vcdimager builds ok with this version (on all arch's) ... (tests for vcdimager are ok on all arch's too) ...

**Comment by Begasus** [comment]
Checked (with same results for the tests), vcdimager builds ok (and tests are ok) on gcc2h (for both arch's), don't have the time to check on x86_64, but I suspect it should be ok ...

**Comment by Begasus** [comment]
Will do, but it won't be for today :)

**Comment by alaviss** [comment]
I've investigated the issue (#2777). The "fix" would be to remove

```c
#define HAVE_REALPATH 1
```

from the generated `config.h` so that it would use the fallback implementation and correctly returns what the library expect.

File: dev-libs/libcdio/libcdio-2.0.0.recipe [hunk 2]
```diff
@@ -0,0 +1,107 @@
+SUMMARY="A library to encapsulate CD-ROM reading and control"
+DESCRIPTION="The libcdio package contains a library for CD-ROM and CD image \
+access. Applications wishing to be oblivious of the OS- and device-dependent \
+properties of a CD-ROM or of the specific details of various CD-image formats \
+may benefit from using this library."
+HOMEPAGE="https://www.gnu.org/software/libcdio/"
+COPYRIGHT="2002-2014 Rocky Bernstein and Herbert Valerio Riedel"
```
Comment (fbrosson) [inline dev-libs/libcdio/libcdio-2.0.0.recipe]: `2002-2014` :arrow_right:  `2002-2017`
(Found 2017 in `ChangeLog`)

File: dev-libs/libcdio/libcdio-2.0.0.recipe [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="http://ftpmirror.gnu.org/libcdio/libcdio-$portVersion.tar.gz"
```
Comment (fbrosson) [inline dev-libs/libcdio/libcdio-2.0.0.recipe]: Suggestion:
```diff
-SOURCE_URI="http://ftpmirror.gnu.org/libcdio/libcdio-$portVersion.tar.gz"
+SOURCE_URI="https://ftpmirror.gnu.org/libcdio/libcdio-$portVersion.tar.gz
+	https://ftp.gnu.org/gnu/libcdio/libcdio-$portVersion.tar.gz"
```

(It also works for `0.94`, but I don't know if it's worth also updating the SOURCE_URI of the old recipe. Well, maybe it is. Do whatever you think is best :)

File: dev-libs/libcdio/libcdio-2.0.0.recipe [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+	"
+if [ -z "$secondaryArchSuffix" ]; then
+	PROVIDES="$PROVIDES
+		cmd:cdda_player
+		cmd:cd_drive
+		cmd:cd_info
+		cmd:cd_read
+		cmd:iso_info
+		cmd:iso_read
+		cmd:mmc_tool
```
Comment (fbrosson) [inline dev-libs/libcdio/libcdio-2.0.0.recipe]: Do we add a ` = $portVersion` for each `cmd:*` entry in here?
(My understanding is that it is recommended to do so, but I guess we can keep this as is.)

Regarding the `libVersiion[Compat]` vars, maybe we should define `libcdioVersion`, `libcdioxxVersion`, `libiso9660Version`, `libiso9660xxVersion` and `libudfVersion` and use the eval "trick" to define the `lib*VersionCompat` the way we do it in e.g. `util_linux`.

File: dev-libs/libcdio/libcdio-2.0.0.recipe [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+libcdioLibs="libcdio libcdioxx libiso9660"
+libVersion="0.0.0"
+libcdioVersion="18.0.0"
+libcdioxxVersion="1.0.0"
+libiso9660Version="11.0.0"
+for i in lib $libcdioLibs; do
+	eval "${i}VersionCompat=\"\$${i}Version compat >= \${${i}Version%%.*}\""
+done
+otherlibcdioLibs="libiso9660++ libudf"
```
Comment (fbrosson) [inline dev-libs/libcdio/libcdio-2.0.0.recipe]: Maybe we should not assume that `libiso9660++.so.0.0.0` and `libudf.so.0.0` will always end with `.0.0.0`.
So, what do you think of this little variant?
```diff
-libcdioLibs="libcdio libcdioxx libiso9660"
+libcdioLibs="
+	libcdio
+	libcdioxx
+	libiso9660
+	libiso9660xx
+	libudf
+	"
-libVersion="0.0.0"
-libcdioVersion="18.0.0"
+libcdioVersion=18.0.0
-libcdioxxVersion="1.0.0"
+libcdioxxVersion=1.0.0
-libiso9660Version="11.0.0"
+libiso9660Version=11.0.0
+libiso9660xxVersion=0.0.0
+libudfVersion=0.0.0
 for i in lib $libcdioLibs; do
 	eval "${i}VersionCompat=\"\$${i}Version compat >= \${${i}Version%%.*}\""
 done
-otherlibcdioLibs="libiso9660++ libudf"
```

Of course, the other comments are part of this variant.
I'm also suggesting a few cosmetic/style changes, but please feel free to ignore them if you wish.

File: dev-libs/libcdio/libcdio-2.0.0.recipe [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+PROVIDES="
+	libcdio$secondaryArchSuffix = $portVersion
+	lib:libcdio$secondaryArchSuffix = $libcdioVersionCompat
+	lib:libcdio++$secondaryArchSuffix = $libcdioxxVersionCompat
+	lib:libiso9660$secondaryArchSuffix = $libiso9660VersionCompat
+	lib:libiso9660++$secondaryArchSuffix = $libVersionCompat
+	lib:libudf$secondaryArchSuffix = $libVersionCompat
```
Comment (fbrosson) [inline dev-libs/libcdio/libcdio-2.0.0.recipe]:
```diff
-	lib:libiso9660++$secondaryArchSuffix = $libVersionCompat
+	lib:libiso9660++$secondaryArchSuffix = $libiso9660xxVersionCompat
-	lib:libudf$secondaryArchSuffix = $libVersionCompat
+	lib:libudf$secondaryArchSuffix = $libudfVersionCompat
```

File: dev-libs/libcdio/libcdio-2.0.0.recipe [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+	lib:libncurses$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	libcdio${secondaryArchSuffix}_devel = $portVersion
+	devel:libcdio${secondaryArchSuffix} = $libcdioVersionCompat
+	devel:libcdio++${secondaryArchSuffix} = $libcdioxxVersionCompat
+	devel:libiso9660${secondaryArchSuffix} = $libiso9660VersionCompat
+	devel:libiso9660++${secondaryArchSuffix} = $libVersionCompat
+	devel:libudf${secondaryArchSuffix} = $libVersionCompat
```
Comment (fbrosson) [inline dev-libs/libcdio/libcdio-2.0.0.recipe]:
```diff
 PROVIDES_devel="
 	libcdio${secondaryArchSuffix}_devel = $portVersion
-	devel:libcdio${secondaryArchSuffix} = $libcdioVersionCompat
+	devel:libcdio$secondaryArchSuffix = $libcdioVersionCompat
-	devel:libcdio++${secondaryArchSuffix} = $libcdioxxVersionCompat
+	devel:libcdio++$secondaryArchSuffix = $libcdioxxVersionCompat
-	devel:libiso9660${secondaryArchSuffix} = $libiso9660VersionCompat
+	devel:libiso9660$secondaryArchSuffix = $libiso9660VersionCompat
-	devel:libiso9660++${secondaryArchSuffix} = $libVersionCompat
+	devel:libiso9660++$secondaryArchSuffix = $libiso9660xxVersionCompat
-	devel:libudf${secondaryArchSuffix} = $libVersionCompat
+	devel:libudf$secondaryArchSuffix = $libudfVersionCompat
 	"
```

File: dev-libs/libcdio/libcdio-2.0.0.recipe [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+debugList=(
+	"$libDir"/libcdio.so.$libcdioVersion
+	"$libDir"/libcdio++.so.$libcdioxxVersion
+	"$libDir"/libiso9660.so.$libiso9660Version
+	)
+for i in $otherlibcdioLibs; do
+	debugList+=("$libDir"/${i/_/-}.so.$libVersion)
+done
+defineDebugInfoPackage libcdio$secondaryArchSuffix "${debugList[@]}"
```
Comment (fbrosson) [inline dev-libs/libcdio/libcdio-2.0.0.recipe]:
```diff
-debugList=(
-	"$libDir"/libcdio.so.$libcdioVersion
-	"$libDir"/libcdio++.so.$libcdioxxVersion
-	"$libDir"/libiso9660.so.$libiso9660Version
-	)
-for i in $otherlibcdioLibs; do
-	debugList+=("$libDir"/${i/_/-}.so.$libVersion)
-done
-defineDebugInfoPackage libcdio$secondaryArchSuffix "${debugList[@]}"
+defineDebugInfoPackage libcdio$secondaryArchSuffix \
+	"$libDir"/libcdio.so.$libcdioVersion \
+	"$libDir"/libcdio++.so.$libcdioxxVersion \
+	"$libDir"/libiso9660.so.$libiso9660Version \
+	"$libDir"/libiso9660++.so.$libiso9660xxVersion \
+	"$libDir"/libudf.so.$libudfVersion \
 
 ```

File: dev-libs/libcdio/libcdio-2.0.0.recipe [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+	make install
+
+	rm $libDir/lib*.la
+
+	if [ -n "$secondaryArchSuffix" ]; then
+		rm -rf $documentationDir $binDir
+	fi
+
+	# set up the develop directory correctly
+	prepareInstalledDevelLibs libcdio libcdio++ libudf libiso9660 libiso9660++
```
Comment (fbrosson) [inline dev-libs/libcdio/libcdio-2.0.0.recipe]:
```diff
 INSTALL()
 {
 	make install
 
-	rm $libDir/lib*.la
+	rm -f "$libDir"/lib*.la
 
 	if [ -n "$secondaryArchSuffix" ]; then
-		rm -rf $documentationDir $binDir
+		rm -rf "$documentationDir" "$binDir"
 	fi
 
 	# set up the develop directory correctly
-	prepareInstalledDevelLibs libcdio libcdio++ libudf libiso9660 libiso9660++
+	prepareInstalledDevelLibs ${libcdioLibs//xx/++}
 	fixPkgconfig
 
 	# devel package
 	packageEntries devel \
-		$developDir
+		"$developDir"
 }
```

File: dev-libs/libcdio/libcdio-2.0.0.recipe (hunk 5 above)
Comment (Begasus) [reply dev-libs/libcdio/libcdio-2.0.0.recipe]: @fbrosson looking at the previous version both libs didn't change libversion, hence why I wanted to pull them into a separate check, started out with the changes you mention here, but I'm ok if you push the changes to the suggested ones :)

Comment (fbrosson) [reply dev-libs/libcdio/libcdio-2.0.0.recipe]: Cool, thanks!

**Comment by fbrosson** [comment]
Shall we drop `libcdio-0.94` in this PR?
(We'll need to rebuild `mpd`, `qmplay2` and `vcdimager`.)

**Comment by Begasus** [comment]
Checked (on gcc2h) for both mpd and vcdimager (vcdimager actually looks for this version), both build ok, I'm ok with dropping the older version :) (providing qmplay2 also buidls ok with this one)

**Comment by alaviss** [comment]
Please make a PR that bumps revision of all users of libcdio.

@fbrosson 

Since the old version isn't co-installable, I think we should just drop it.

**Comment by Begasus** [comment]
Atleast one isn't using this one: https://github.com/haikuports/haikuports/blob/086ace1fa5a980f873f8c5f6c98068368ea4ddbb/media-plugins/audacious-plugins/audacious_plugins-3.10.recipe
It's actually looking for libcdio-paranoia ...

**Comment by fbrosson** [comment]
> Since the old version isn't co-installable, I think we should just drop it.

:+1:
@alaviss: Yep, that's the reason why I'd like to drop it in this PR.

**Comment by Begasus** [comment]
@fbrosson since you pushed your changes into this PR, could you also remove the old recipe then? (if thing are ok) Don't wont to push the removal and re-enable things you changed since then :)

**Comment by korli** [comment]
Please consider renaming libcdio-0.94.recipe for compatibility. That way you don't have to worry right away about rebuilds, and it keeps third party repositories happy.

**Comment by fbrosson** [comment]
Good idea. Let me replace the last commit ("libcdio: drop old recipe") by a "libcdio: rename libcdio-0.94.recipe to libcdio0-0.94.recipe".
If anyone thinks another name would be better then please do not hesitate to suggest alternatives.

BTW, I'll add the appropriate conflicts, of course.

**Comment by fbrosson** [comment]
Done, but this is a blind edit. I did not try to build `libcdio0-0.94`.

**Pull request merged**